test/ibmq/test_ibmq_job.py

```diff
@@ -296,6 +295,41 @@ def test_retrieve_jobs_status(self, provider):
         for job in job_list:
             self.assertTrue(job.status() is JobStatus.DONE)
 
+    @requires_provider
+    def test_retrieve_jobs_queued(self, provider):
+        """Test retrieving jobs that are queued."""
+        backend = most_busy(provider.backends(simulator=False))
+        qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+        job = backend.run(qobj)
+
+        # Wait for the job to queue, run, or reach a final state.
+        leave_states = list(JOB_FINAL_STATES) + [JobStatus.QUEUED, JobStatus.RUNNING]
+        while job.status() not in leave_states:
+            time.sleep(0.5)
+
+        before_status = job.status()
```

Comment:
nit - you can use `job._status` here since you just called `job.status()` above. While it's not much, it saves 1 trip to the api and therefore reduces the chance the job may transition to RUNNING.